### PR TITLE
Repositories with alternative certificate and catalog path

### PR DESCRIPTION
**Pull request by jblomer**
Not quite ready yet but close.  While working on this prerequisite for CVM-904, I realized that a fixed path for the symlink (like .cvmfsroot) has a race condition: the time between updating the symlink and updating .cvmfspublished.  The symlink must contain the hash.  Of course, we accumulate symlinks this way but this is a second order problem which can be easily solved.

### CODE REVIEW

**Comment by jblomer** [comment]
Ready to merge from my side.  

Eventually, we probably want to have an alternative path for the history database, too.  We also might want to add support for alternative paths into the replication code.  But I think we don't need to care about this right now.

@bbockelm please let me know if you have any comments.

File: cvmfs/manifest.cc [hunk 1]
```diff
@@ -114,7 +118,8 @@ string Manifest::ExportString() const {
     "R" + root_path_.ToString() + "\n" +
     "D" + StringifyInt(ttl_) + "\n" +
     "S" + StringifyInt(revision_) + "\n" +
-    "G" + StringifyBool(garbage_collectable_) + "\n";
+    "G" + StringifyBool(garbage_collectable_) + "\n"
```
Comment (reneme) [inline cvmfs/manifest.cc]: Knit picking: That is missing a `+` in the end of the line. Doesn't matter for compilation, but maybe for the sake of consistency?

File: cvmfs/upload_local.cc [hunk 2]
```diff
@@ -215,7 +236,24 @@ void LocalUploader::FinalizeStreamedUpload(UploadStreamHandle  *handle,
...
+               "(errno=%d)", final_path.c_str(), dest_path.c_str(),
+               strerror(errno), errno);
+      atomic_inc32(&copy_errors_);
+      Respond(handle->commit_callback, UploaderResults(cpy_errno));
+      return;
+    }
+  }
+
   const CallbackTN *callback = handle->commit_callback;
+  // TODO(jblomer): do we leak memory if the function is left earlier?
```
Comment (reneme) [inline cvmfs/upload_local.cc]: we do! But if it is left early, the using code bails out with an `assert()`...

Comment (reneme) [inline cvmfs/upload_local.cc]: ... TODO is reasonable though. ;-)

File: cvmfs/swissknife_sync.cc [hunk 3]
```diff
@@ -595,7 +600,8 @@ int swissknife::CommandSync::Main(const swissknife::ArgumentList &args) {
   sync->Traverse();
 
   LogCvmfs(kLogCvmfs, kLogStdout, "Exporting repository manifest");
-  UniquePtr<manifest::Manifest> manifest(mediator.Commit());
+  const bool use_alt_path = params.voms_authz.size();
```
Comment (reneme) [inline cvmfs/swissknife_sync.cc]: please! 0 and 1 are somewhat boolean, but why not `! params.voms_authz.empty()`?

File: cvmfs/manifest.cc (hunk 1 above)
Comment (jblomer) [inline cvmfs/manifest.cc]: Yep, `+` is missing

File: cvmfs/swissknife_sync.cc (hunk 3 above)
Comment (jblomer) [inline cvmfs/swissknife_sync.cc]: Yep, `empty()` is better.  Please stay positive in these reviews.

**Comment by jblomer** [comment]
Superseded by #1291

**Pull request closed without merging**